frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/new-experiment/new-experiment.component.ts

```diff
@@ -49,6 +51,19 @@ export class NewExperimentComponent implements OnInit {
           ...this.newExperimentData,
           ...formData
         };
+
+        if (!this.currentContext && this.experimentInfo) {
+          this.currentContext = this.experimentInfo.context[0];
+        }
+
+        this.isContextChanged = this.currentContext ? (this.currentContext !== this.newExperimentData.context[0]) : false;
+
+        if (this.experimentInfo) {
+          this.currentContext = this.experimentInfo.context[0]
```

Comment:
We don't need this logic at all as the value of `this.currentContext` is changed after the if condition always.

Reply:
code updated